### PR TITLE
Added COIL-100 dataset

### PR DESCRIPTION
**Pull request by SoumyadeepJana**

### CODE REVIEW

**Comment by SoumyadeepJana** [comment]
https://github.com/tensorflow/datasets/issues/262
@Conchylicultor @cyfra Kindly review :)

File: tensorflow_datasets/image/coil100.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+The turntable was rotated through 360 degrees to vary object pose with respect to a fxed color camera.
+Images of the objects were taken at pose intervals of	5 degrees.This corresponds to
+72 poses per object""")
+
+_LABELS = [str(x) for x in range(0,360,5)]
+
+_IMAGE_SHAPE = (128,128,3)
+
+_CITATION = """\
+
```
Comment (us) [inline tensorflow_datasets/image/coil100.py]: unnecessary empty line

**Comment by Conchylicultor** [comment]
Thank you for all your contributions. We would really like to see all those datasets added in tfds.
However, it seems that you are not following our style guide: https://github.com/tensorflow/datasets/blob/master/docs/add_dataset.md#5-check-your-code-style

Could you run `./oss_scripts/lint.sh` on all your PR and update them accordingly ?

File: tensorflow_datasets/image/coil100.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  """COIL-100 Image Dataset Class"""
+
+  VERSION = tfds.core.Version('1.0.0')
+
+  def _info(self):
+        """Define Dataset Info"""
+
+        return tfds.core.DatasetInfo(
+            builder=self,
+
```
Comment (Conchylicultor) [inline tensorflow_datasets/image/coil100.py]: Why the extra line here ? It seems you have this in all your PR ?

Comment (SoumyadeepJana) [reply tensorflow_datasets/image/coil100.py]: Yes, I missed those standard specifications, sorry for that. I will fix them ASAP.

**Comment by SoumyadeepJana** [comment]
@Conchylicultor Implemented suggested changes.Kindly Review :)

File: tensorflow_datasets/image/coil100.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+  def _info(self):
+    """Define Dataset Info"""
+
+    return tfds.core.DatasetInfo(
+        builder=self,
+        description=(_DESCRIPTION),
+        features=tfds.features.FeaturesDict({
+            "image": tfds.features.Image(shape=_IMAGE_SHAPE),
+            "label": tfds.features.ClassLabel(names=_LABELS),
+            "object_id": tfds.features.Text()
```
Comment (Conchylicultor) [inline tensorflow_datasets/image/coil100.py]: Shouldn't the object_id label be the class instead ?
People training model on this dataset probably want to predict the object class from the image.
The orientation should be given as is as integer.

So you would have something like:

```python
{
            "image": tfds.features.Image(shape=_IMAGE_SHAPE),
            "label": tfds.features.ClassLabel(names=['obj{}'.format(i) for i in range(1, 101)]),
            "orientation": tf.int64,
}
```

**Comment by Conchylicultor** [comment]
When I generate the dataset, I get empty tf-records (you can see this in the `dataset_info.json` which should contains the statistics and number of examples). Here it seems empty

File: tensorflow_datasets/image/coil100.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        }),
+        supervised_keys=("image", "label"),
+        urls=[_URL],
```
Comment (Conchylicultor) [inline tensorflow_datasets/image/coil100.py]: The url here should contains the homepage of the dataset, not the download url zip file. Could you update it please ?

**Comment by us** [comment]
@SoumyadeepJana hey still continue?

**Comment by cyfra** [comment]
Migrated to #825

**Pull request closed without merging**